Content.Server/Chemistry/ReagentEffects/RemoveComponent.cs

```diff
@@ -0,0 +1,48 @@
+using Content.Shared.Chemistry.Reagent;
+using Content.Server.Medical;
+using JetBrains.Annotations;
+using Robust.Shared.Prototypes;
+using Robust.Shared.GameObjects;
+using Robust.Shared.IoC;
+using System.ComponentModel;
+using System.Runtime.CompilerServices;
+using Robust.Shared.IoC;
+
+namespace Content.Server.Chemistry.ReagentEffects
+{
+    /// <summary>
+    /// Removes designated component.
+    /// You can apply this effect to the same reagent multiple times for multiple components and it'll still work.
+    /// </summary>
+    [UsedImplicitly]
+    public sealed class RemoveComponent : ReagentEffect
+    {
+        /// <summary>
+        /// Name of component to remove, as a string
+        /// Note: component name shouldn't have the "component" postfix
+        /// WRONG: [component: ReplacementAccentComponent]
+        /// RIGHT: [component: ReplacementAccent]
+        /// </summary>
+        [DataField("component")]
+        public string Component = String.Empty;
```

Comment:
Use componentregistry so it serializes correctly.

Reply:
Looking at other instances of people doing what I'm trying to do (primarily PR 13125), it seems that using a HashSet is a better option. I tried a ComponentRegistry but it seems to be tailored more for adding components than removing them, judging by a 'required field not implemented' error (as ReplacementAccentComponent has an "accent" field marked as "required"). I've implemented a HashSet, but I'm not experienced enough to judge whether it'll serialize correctly or not.

Reply:
So anyways, I have no idea how to tell if something will serialize and so in light of recent advice I'll just ask: is the current implementation good in terms of serializing?

Reply:
I'll just resolve this, please make a new comment if there are problems

Reply:
ComponentRegistry just stores component types, it's fine to use for whatever. You should be able to just grab the type from it and use RemoveComponent(EntityUid, Type).